pkg/beaconfqdn/writer.go

```diff
@@ -44,44 +45,38 @@ func (w *writer) close() {
 //start kicks off a new write thread
 func (w *writer) start() {
 	w.writeWg.Add(1)
-
 	go func() {
 		ssn := w.db.Session.Copy()
 		defer ssn.Close()
 
-		for data := range w.writeChannel {
+		bulk := ssn.DB(w.db.GetSelectedDB()).C(w.targetCollection).Bulk()
+		bulk.Unordered()
+		count := 0
 
-			if data.beacon.query != nil {
-				// update beacons table
-				info, err := ssn.DB(w.db.GetSelectedDB()).C(w.targetCollection).Upsert(data.beacon.selector, data.beacon.query)
+		for data := range w.writeChannel {
+			bulk.Upsert(data.selector, data.query)
+			count++
 
-				if err != nil ||
-					((info.Updated == 0) && (info.UpsertedId == nil)) {
+			if count >= 1000 {
+				info, err := bulk.Run()
```

Comment:
The bulk upserts that are run here are sometimes causing Mongo to error because part or all of the upsert statement provided is over 16MB in size. When enough of these errors occur, it causes the entire MongoDB server to crash.

Attached are the RITA error logs. 
[error-bulk.txt](https://github.com/activecm/rita/files/8969150/error-bulk.txt)

Reply:
Good catch! I'll cut the limit in half with a comment explaining why. We could probably get a way with 900, but I think 500 gets us a little more head room.

Reply:
Pushed a fix